components/post_view/post_attachment.jsx

```diff
@@ -35,9 +38,44 @@ export default class PostAttachment extends React.PureComponent {
 
         this.state = {
             collapsed: true,
+            hasOverflow: false,
         };
     }
 
+    componentDidMount() {
+        this.checkAttachmentTextOverflow();
+
+        window.addEventListener('resize', this.handleResize);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.attachment.text !== prevProps.attachment.text) {
+            this.checkAttachmentTextOverflow();
+        }
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.handleResize);
+    }
+
+    checkAttachmentTextOverflow = () => {
```

Comment:
This also needs to be called whenever a Markdown image loads its height, otherwise loading a tall image after the post renders may not trigger the overflow behaviour. You can look at [PostMessageView](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/post_message_view/post_message_view.jsx) to see how it stores `this.imageProps` and passes it into the Markdown component later on

Reply:
Oh I see, got it, thanks!
Now updated, please have a second look again before I merge this one.